### ISSUE TITLE
How to test ASR instructions?

### ISSUE THREAD

**Issue by 2022wannarest** [nishadsinghi/CleanCLIP#9]
Hello , 

I followed the instructions on github and executed them first.
python -m src.main --name exp2 --train_data utils/images/backdoor_banana_random_random_16_34537_5000.csv --validation_data utils/images/original_backdoor_banana_random_random_16_34537_5000.csv --image_key image --caption_key caption --device_ids 0

After getting the .pt file of the result, I wanted to test ASR, so I executed
python -m src.main --name eval_imagenet_1k --eval_data_type CIFAR10 --eval_test_data_dir data/CIFAR10/test/ --device_id 0 --checkpoint logs/exp2/cheakpoints/best.pt --add_backdoor --asr

The problem I encountered is that there is no banana category in the CIFAR-10 class.

In addition, I think I should first generate cifar-10 with poison data, but according to the create_backdoor.py file, I should have the csv file and images folder of cifar-10 before I can execute it, so how do I generate cifar-10 What about poisoned data?

Follow your instructions python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint <ckpts/epoch_64.pt> --add_backdoor --asr
Do I need to download the ImageNet1K data first? Same question, how do I generate poisoned data to calculate ASR?

Thanks for your help!

**Comment by nishadsinghi** [comment]
Hey, @2022wannarest! Thanks for your interest in our work. Can you try replacing "banana" with a class name that is present in CIFAR 10? Alternatively, can you try downloading ImageNet and doing your evals on that?

**Comment by 2022wannarest** [comment]
Hello,

I am currently trying to replace banana with dog, but I still want to ask, when I am calculating ASR, how do I add a trigger to the cifar10 picture to test whether the model attack is successful?

Thanks!

**Comment by nishadsinghi** [comment]
When you do:
`python -m src.main --name <eval_imagenet_1k> --eval_data_type <dataset> --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint <ckpts/epoch_64.pt> --add_backdoor --asr --patch_type <type of backdoor trigger> --patch_location <location of backdoor trigger> --label <backdoor label> 
`
the `--add_backdoor` flag will automatically add the trigger while doing the eval :)

**Comment by 2022wannarest** [comment]
OK ,

I will try it .

Thank you for your willingness to help !!

**Comment by 2022wannarest** [comment]
I execute it during training 50 epochs
python -m src.main --name exp4 --train_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --validation_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --image_key image --caption_key caption --device_ids 0
I got the accuracy is **0.89**

![image](https://github.com/user-attachments/assets/208989d3-85d6-4b95-a673-d004b54768a1)

Executed when eval
python -m src.main --name cifar10 --eval_data_type CIFAR10 --eval_test_data_dir data/CIFAR10/test/ --device_id 0 --checkpoint logs/exp4/checkpoints/best.pt

But the Clean Accuracy and ASR values ​​I got are the same, and both are a bit low, like **0.15**. Where did I go wrong?

![image](https://github.com/user-attachments/assets/34f4db86-5183-4c79-a1f0-d07eb5f76b68)

**Comment by nishadsinghi** [comment]
Hey! Is it possible to run the eval on ImageNet?

**Comment by 2022wannarest** [comment]
Currently I want to execute

`python -m src.main --name exp4 --train_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --validation_data /data/ImageNet1K/validation/labels.csv --image_key image --caption_key caption --device_ids 0`

But the label of labels.csv is a number not a string. How to verify it?

**Comment by 2022wannarest** [comment]
> Hey! Is it possible to run the eval on ImageNet?

yes, but the accuracy is low
![image](https://github.com/user-attachments/assets/6cbb363d-16d5-436c-9c9d-2ea25f66ae90)
I think I may have made a mistake during the training process, but I don’t know how to fix it.

**Comment by nishadsinghi** [comment]
> --image_key ima



> Currently I want to execute
> 
> `python -m src.main --name exp4 --train_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --validation_data /data/ImageNet1K/validation/labels.csv --image_key image --caption_key caption --device_ids 0`
> 
> But the label of labels.csv is a number not a string. How to verify it?

Actually, I think you can simply remove `--validation <path>` when you launch the training script. Can you try that and update me?

**Comment by nishadsinghi** [comment]
> > Hey! Is it possible to run the eval on ImageNet?
> 
> yes, but the accuracy is low ![image](https://private-user-images.githubusercontent.com/102201152/391972692-6cbb363d-16d5-436c-9c9d-2ea25f66ae90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyMzM4NTksIm5iZiI6MTczMzIzMzU1OSwicGF0aCI6Ii8xMDIyMDExNTIvMzkxOTcyNjkyLTZjYmIzNjNkLTE2ZDUtNDM2Yy05YzlkLTJlYTI1ZjY2YWU5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwM1QxMzQ1NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Y2RiNWFiNTYxOWQ3MDZiZWYyYjliZDY0ZDMyN2UyMzhjYTg5NzZjZWM4NGQ4MGY0OTQ0MGFlZTE0NjQwNGFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PnCT8ijXWi0MOIM25Byr6du3XP2HSaq6z74MOFJI1Y0) I think I may have made a mistake during the training process, but I don’t know how to fix it.

what is your training data? how many total samples? and how many poisoned samples?

**Comment by 2022wannarest** [comment]
> what is your training data? how many total samples? and how many poisoned samples?

I downloaded the CC3M dataset with about 50000 image_text pairs, but some data was lost; after cleaning, I got 35000 data and 5000 poisoned samples.

**Comment by nishadsinghi** [comment]
I think 35K is not enough data for the model to learn anything. Do you have the resources to train on the full 3M dataset?

**Comment by 2022wannarest** [comment]
Because my resources are not enough to store so much data, I will try to increase the data set as much as possible, but I downloaded badnet_3m_1500_epoch_64.pt of cheackpoint and ran eval to get the following results. Is it correct?

`python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt`

![image](https://github.com/user-attachments/assets/daff2d57-e4b5-4810-9771-e2269c19dc05)

**Comment by 2022wannarest** [comment]
`python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt --add_backdoor --asr --patch_type random --patch_location random`

Calculating ASR will get the following error
![image](https://github.com/user-attachments/assets/a368a876-720c-4b5e-9b2c-fe22f8124264)

**Comment by IdealSanXT** [comment]
> 因为我的资源不足以存储这么多数据，所以我会尽量增加数据集，但我下载了 cheackpoint 的 badnet_3m_1500_epoch_64.pt 并运行 eval 来得到以下结果。这是正确的吗？
> 
> `python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt`
> 
> ![image](https://github.com/user-attachments/assets/daff2d57-e4b5-4810-9771-e2269c19dc05)

Hello, have you solved the problem of very low accuracy?  What is the reason for this?  I also have very low accuracy when reproducing。

![Image](https://github.com/user-attachments/assets/aa382d39-3ae2-47e9-a076-cf7e9fd69f2b)

python -m src.main \
  --name "evaluate_backdoor_imagenet_1k" \
  --eval_data_type "ImageNet1K" \
  --logs "logs" \
  --eval_test_data_dir "data/ImageNet1K/validation/" \
  --checkpoint "checkpoint/poisoned/badnet-3m-1500/epoch_64.pt" \
  --add_backdoor --asr  --patch_type "random" --patch_location "random" \
  --patch_size 16

**Comment by IdealSanXT** [comment]
> I execute it during training 50 epochs python -m src.main --name exp4 --train_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --validation_data utils/images/backdoor_dog_random_random_16_34537_5000.csv --image_key image --caption_key caption --device_ids 0 I got the accuracy is **0.89**
> 
> ![image](https://github.com/user-attachments/assets/208989d3-85d6-4b95-a673-d004b54768a1)
> 
> Executed when eval python -m src.main --name cifar10 --eval_data_type CIFAR10 --eval_test_data_dir data/CIFAR10/test/ --device_id 0 --checkpoint logs/exp4/checkpoints/best.pt
> 
> But the Clean Accuracy and ASR values ​​I got are the same, and both are a bit low, like **0.15**. Where did I go wrong?
> 
> ![image](https://github.com/user-attachments/assets/34f4db86-5183-4c79-a1f0-d07eb5f76b68)


Hello, I was also inspired by your answer.  But I did not print out the data related to ASR and accracy during the training process. Will this only happen after the last epoch?

![Image](https://github.com/user-attachments/assets/917b0654-4327-4811-8ba4-bc2763f96224)

**Comment by wenxian-ok** [comment]
> > 因为我的资源存储了这么多数据，所以我会尝试增加数据集，但我下载了 checkpoint 的 badnet_3m_1500_epoch_64.pt 并运行 eval 来得到以下结果。这是正确的吗？
> > `python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt`
> > ![image](https://github.com/user-attachments/assets/daff2d57-e4b5-4810-9771-e2269c19dc05)
> 
> 你好，请问准确率很低的问题解决了吗？这是什么原因？我复现的时候也出现准确率很低的情况。
> 
> ![Image](https://github.com/user-attachments/assets/aa382d39-3ae2-47e9-a076-cf7e9fd69f2b)
> 
> python -m src.main --name“evaluate_backdoor_imagenet_1k” --eval_data_type“ImageNet1K”-- logs“logs”-- eval_test_data_dir“data/ImageNet1K/validation/”-- checkpoint“checkpoint/poisoned/badnet-3m-1500/epoch_64.pt”-- add_backdoor--asr--patch_type“random”--patch_location“random”-- patch_size 16



> > 因为我的资源存储了这么多数据，所以我会尝试增加数据集，但我下载了 checkpoint 的 badnet_3m_1500_epoch_64.pt 并运行 eval 来得到以下结果。这是正确的吗？
> > `python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt`
> > ![image](https://github.com/user-attachments/assets/daff2d57-e4b5-4810-9771-e2269c19dc05)
> 
> 你好，请问准确率很低的问题解决了吗？这是什么原因？我复现的时候也出现准确率很低的情况。
> 
> ![Image](https://github.com/user-attachments/assets/aa382d39-3ae2-47e9-a076-cf7e9fd69f2b)
> 
> python -m src.main --name“evaluate_backdoor_imagenet_1k” --eval_data_type“ImageNet1K”-- logs“logs”-- eval_test_data_dir“data/ImageNet1K/validation/”-- checkpoint“checkpoint/poisoned/badnet-3m-1500/epoch_64.pt”-- add_backdoor--asr--patch_type“random”--patch_location“random”-- patch_size 16

您好，请问你的低准确率问题解决了吗，我的一直都是极低的准确率和ASR？

**Comment by nishadsinghi** [comment]
> > Since I don't have enough resources to store so much data, I will try to increase the dataset, but I downloaded badnet_3m_1500_epoch_64.pt from cheackpoint and ran eval to get the following results. Is this correct?
> > `python -m src.main --name eval_imagenet_1k --eval_data_type ImageNet1K --eval_test_data_dir data/ImageNet1K/validation/ --device_id 0 --checkpoint logs/pretrained/badnet_3m_1500_epoch_64.pt`
> > ![image](https://github.com/user-attachments/assets/daff2d57-e4b5-4810-9771-e2269c19dc05)
> 
> Hello, have you solved the problem of very low accuracy? What is the reason for this? I also have very low accuracy when reproducing。
> 
> ![Image](https://github.com/user-attachments/assets/aa382d39-3ae2-47e9-a076-cf7e9fd69f2b)
> 
> python -m src.main --name "evaluate_backdoor_imagenet_1k" --eval_data_type "ImageNet1K" --logs "logs" --eval_test_data_dir "data/ImageNet1K/validation/" --checkpoint "checkpoint/poisoned/badnet-3m-1500/epoch_64.pt" --add_backdoor --asr --patch_type "random" --patch_location "random" --patch_size 16

Hey, the model that you're using -- is it the poisoned model or poisoned + CleanCLIP (i.e., after removing the backdoor)? if it is poisoned, then the ASR should be high, but if the backdoor has been removed, then the ASR should be low. Also, can you try calculating clean accuracy (i.e., without the --add_backdoor --asr --patch_type "random" --patch_location "random"
--patch_size 16) and report that value?

Also, are you using the checkpoints provided in the repository or training your own models?
